source/mir/math/stat.d

```diff
@@ -220,7 +239,162 @@ struct MeanAccumulator(T, Summation summation)
     void put()(T x)
     {
         count++;
-        sumAccumulator.put(x);
+        summator.put(x);
+    }
+    
+    ///
+    void put(F, MeanAlgo algo, Summation sumalgo)(MeanAccumulator!(F, algo, sumalgo) m)
+        if (algo == MeanAlgo.precise)
+    {
+        count += m.count;
+        summator.put(cast(T) m.summator.sum);
+    }
```

Comment:
A summator should accept the same summation type directly, not the sum. Accepting mean with other sumalgo seems like overengineering.